### ISSUE TITLE
spring boot admin support

### ISSUE THREAD

**Issue by SankM1** [openzipkin/zipkin#3739]
Able to register on eureka and also able to see actuator endpoints as it is fixed in 3.1.0 release but zipkin is not getting registered on spring boot admin.

I have registered 2 services along with zipkin on eureka ,

![eureka](https://github.com/openzipkin/zipkin/assets/156760529/c731a47b-1b5f-457e-9add-809375f6687d)

But on spring boot admin only 2 services are visible , zipkin is not getting registered on admin

![admin](https://github.com/openzipkin/zipkin/assets/156760529/4150ee8d-13d6-415f-824a-ea451bb5e400)

Kindly check, Thank you.

**Comment by codefromthecrypt** [comment]
so we never said we supported spring boot admin, but I'll take a look and see what we can do.

**Comment by codefromthecrypt** [comment]
one thing I would like to clarify is that while zipkin uses spring boot internally, there is intentionally no promises around spring boot compatibility. This allows us to upgrade boot without impact and also a choice each major version, as to whether to continue to implement our server with it or not. In other words, if this UI requires something very specific to spring boot, we may choose to not support this.

**Comment by codefromthecrypt** [comment]
ok I looked into the problem of zipkin not showing up in spring-boot-admin despite configuring eureka. My assumption is that anything in eureka not filtered out by service, should show up.

I started eureka, zipkin then SpringBootAdminEurekaApplication and used a debugger and debug logging.

I saw zipkin's service/app go by, but no instances returned.

https://github.com/codecentric/spring-boot-admin/blob/master/spring-boot-admin-server-cloud/src/main/java/de/codecentric/boot/admin/server/cloud/discovery/InstanceDiscoveryListener.java#L126-L128

It appears that `discoveryClient.getServices` is flat mapped to `discoveryClient::getInstances`.. While this sounds intuitive, it won't work with the spring-cloud-netflix code. Basically getInstances does *not* take a service name, and in the case of eureka it is a VIP address. I recently found a [7 year old issue](https://github.com/spring-cloud/spring-cloud-netflix/issues/1788) on this for exactly the same reason.

I doubt spring-cloud-netflix will ever change/fix this mismatch, so if you want to do what looks like is being attempted, spring-boot-admin would need to use the underlying eureka client and not spring cloud discovery to do it.

For example, I don't think Eureka has a way to get service/app names without getting everything anyway, so the easiest path could be to just call the /eureka/v2/apps endpoint instead of flat mapping around https://github.com/Netflix/eureka/wiki/Eureka-REST-operations

cc @joshiste @joshiste and @spencergibb in case I'm wrong on any of the above.

**Comment by codefromthecrypt** [comment]
ps the only way to get this to work without changing anything, is to have the zipkin instance register its vipAddress to the same name as its service.

For example, in docker, if you set the container name to "zipkin" and set the env "EUREKA_HOSTNAME=zipkin", it would probably work.

**Comment by codefromthecrypt** [comment]
TL;DR; set EUREKA_HOSTNAME=zipkin (and make sure that's a valid hostname) as it is the only currently known way to be compatible with spring-cloud-netflix discovery impl for eureka. Feel free to make more suggestions after close!

Yep, that was the issue. I replaced the docker-compose.yaml [here](https://github.com/codecentric/spring-boot-admin/tree/master/spring-boot-admin-samples/spring-boot-admin-sample-eureka) and built and ran. worked as expected.

```yaml
version: '2.4'

services:
  eureka:
    image: ghcr.io/openzipkin/zipkin-eureka:${TAG:-latest}
    container_name: eureka

  zipkin:
    image: ghcr.io/openzipkin/zipkin-slim:${TAG:-latest}
    container_name: zipkin
    environment:
      - EUREKA_SERVICE_URL=http://eureka:8761/eureka/v2
      - EUREKA_HOSTNAME=zipkin
    depends_on:
      eureka:
        condition: service_healthy

  admin:
    build:
      context: .
      dockerfile: ./src/main/docker/Dockerfile
    container_name: admin
    ports:
     - "8080:8080"
    environment:
      - EUREKA_SERVICE_URL=http://eureka:8761
    depends_on:
      eureka:
        condition: service_healthy
```

<img width="1089" alt="Screenshot 2024-02-28 at 22 28 44" src="https://github.com/openzipkin/zipkin/assets/64215/f06c0c59-19a9-47ad-9ee0-45d05f140957">
